### PR TITLE
fix: Remove premature deserialization of settings in the CLI

### PR DESCRIPTION
**Pull request by edgarrmondragon**
See in Slack https://meltano.slack.com/archives/C03GKHWS0HM/p1669128539584859

### CODE REVIEW

File: src/meltano/cli/interactive/config.py [hunk 1]
```diff
@@ -368,9 +365,6 @@ def configure_all(self):
 
     def set_value(self, setting_name, value, store, interactive=False):
         """Set value helper function."""
-        with contextlib.suppress(json.JSONDecodeError):
-            value = json.loads(value)
```
Comment (WillDaSilva) [inline src/meltano/cli/interactive/config.py:372]: @edgarrmondragon @kgpayne Why was this here in the first place?

**Review by tayloramurphy — APPROVED** [review]
Approving - I absolutely _love_ these integration examples. Super clear what's happening.

File: src/meltano/cli/interactive/config.py (hunk 1 above)
Comment (kgpayne) [reply src/meltano/cli/interactive/config.py:372]: It was added by this [MR](https://gitlab.com/meltano/meltano/-/merge_requests/1764/diffs) back in Gitlab world. We do accept json-formatted strings using `meltano set` for (object and array types), so this was likely added to solve for that use case. As @edgarrmondragon mentions, we do cast by kind at a later stage now, so this may just be an older artefact 🤷‍♂️

Comment (edgarrmondragon) [reply src/meltano/cli/interactive/config.py:372]: @WillDaSilva I found https://github.com/meltano/meltano/commit/6c912d65e58017c781df212bba1b464b29a12ebb and https://gitlab.com/meltano/meltano/-/merge_requests/1764/.

Though that work was done after `object` and `array` were introduced https://github.com/meltano/meltano/commit/fcda3c948c572a28c16dd09a2480a4bfc0f1aae4 and https://github.com/meltano/meltano/commit/12362f93f01dcacb54251e5d18c2928f55c12dd9.

So, I'm gonna do some testing with `meltano config ... set` for schemas to see if anything's off.

Comment (edgarrmondragon) [reply src/meltano/cli/interactive/config.py:372]: Setting the `_schema` extra works fine too, as any normal `object` 🤔

@kgpayne @WillDaSilva it seems safe to merge. Wdyt?

**Review by kgpayne — APPROVED** [review]

File: src/meltano/cli/interactive/config.py (hunk 1 above)
Comment (kgpayne) [reply src/meltano/cli/interactive/config.py:372]: LGTM 🚢

**Review by WillDaSilva — APPROVED** [review]
